src/Symfony/Component/Security/Core/Validator/Constraints/UserPasswordValidator.php

```diff
@@ -39,6 +39,10 @@ public function validate($password, Constraint $constraint)
             throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\UserPassword');
         }
 
+        if (null === $password || '' === $password) {
+            return;
```

Comment:
The [implementation for 2.7 looks different](https://github.com/symfony/security-core/blob/2.7/Encoder/BasePasswordEncoder.php#L84).